### PR TITLE
Dynamically generate IdP creation wizard from templates 

### PR DESCRIPTION
**Pull request by tharindu-bandara**
## Purpose
- Addresses https://github.com/wso2/identity-apps/issues/354.

## Goals
- Generate wizard steps based on the template.

## Approach
- Steps are initialized with the template attributes and values.
Screenshot from 2020-04-01 19-47-36
- General settings and Summary steps are common for every template. The rest of the supported steps are generated according to the template. For example, if the Authenticator step is not available in the template, that will be excluded from wizard steps.
Screenshot from 2020-04-01 19-50-59
- And also, the Summary page is rendered accordingly.
Screenshot from 2020-04-01 19-51-47

### CODE REVIEW

File: apps/admin-portal/src/components/identity-providers/wizard/identity-provider-create-wizard.tsx [hunk 1]
```diff
@@ -329,49 +329,121 @@ export const IdentityProviderCreateWizard: FunctionComponent<IdentityProviderCre
...
+            STEPS = [
+                ...STEPS,
+                {
+                    icon: IdentityProviderWizardStepIcons.authenticator,
+                    name: WizardSteps.AUTHENTICATOR,
+                    submitCallback: setSubmitAuthenticator,
+                    title: "Authenticator"
+                }
+            ];
+
```
Comment (NipuniBhagya) [inline apps/admin-portal/src/components/identity-providers/wizard/identity-provider-create-wizard.tsx:390]: Shall we remove this line?

File: apps/admin-portal/src/components/identity-providers/wizard/steps/general-settings.tsx [hunk 2]
```diff
@@ -16,7 +16,7 @@
  * under the License.
  */
 
-import React, { FunctionComponent } from "react";
+import React, {FunctionComponent, useEffect} from "react";
```
Comment (NipuniBhagya) [inline apps/admin-portal/src/components/identity-providers/wizard/steps/general-settings.tsx:19]:
```suggestion
import React, { FunctionComponent, useEffect } from "react";
```

File: apps/admin-portal/src/components/identity-providers/wizard/steps/wizard-summary.tsx [hunk 3]
```diff
@@ -82,26 +82,28 @@ export const WizardSummary: FunctionComponent<WizardSummaryProps> = (
...
-                            <Heading size="small" className="name">{identityProvider.name}</Heading>
-                        )}
-                        {identityProvider?.description && (
-                            <div className="description">{identityProvider.description}</div>
-                        )}
-                    </div>
-                </Grid.Column>
-            </Grid.Row>
+    const getGeneralDetailsComponent = () => {
+        return <Grid.Row>
```
Comment (NipuniBhagya) [inline apps/admin-portal/src/components/identity-providers/wizard/steps/wizard-summary.tsx:86]: Shall we get the <Grid.Row> to the next line?

File: apps/admin-portal/src/components/identity-providers/wizard/steps/wizard-summary.tsx [hunk 4]
```diff
@@ -82,26 +82,28 @@ export const WizardSummary: FunctionComponent<WizardSummaryProps> = (
...
+            <Grid.Column mobile={ 16 } tablet={ 16 } computer={ 16 } textAlign="center">
+                <div className="general-details">
+                    <AppAvatar
+                        name={ identityProvider?.name }
+                        image={ identityProvider?.image }
+                        size="tiny"
+                    />
+                    {identityProvider?.name && (
+                        <Heading size="small" className="name">{identityProvider.name}</Heading>
```
Comment (NipuniBhagya) [inline apps/admin-portal/src/components/identity-providers/wizard/steps/wizard-summary.tsx:95]:
```suggestion
                        <Heading size="small" className="name">{ identityProvider.name }</Heading>
```

File: apps/admin-portal/src/components/identity-providers/wizard/steps/wizard-summary.tsx [hunk 5]
```diff
@@ -82,26 +82,28 @@ export const WizardSummary: FunctionComponent<WizardSummaryProps> = (
...
+                    )}
+                    {identityProvider?.description && (
+                        <div className="description">{identityProvider.description}</div>
```
Comment (NipuniBhagya) [inline apps/admin-portal/src/components/identity-providers/wizard/steps/wizard-summary.tsx:98]:
```suggestion
                        <div className="description">{ identityProvider.description }</div>
```

File: apps/admin-portal/src/components/identity-providers/wizard/steps/wizard-summary.tsx [hunk 6]
```diff
@@ -82,26 +82,28 @@ export const WizardSummary: FunctionComponent<WizardSummaryProps> = (
...
+                    )}
+                </div>
+            </Grid.Column>
+        </Grid.Row>;
```
Comment (NipuniBhagya) [inline apps/admin-portal/src/components/identity-providers/wizard/steps/wizard-summary.tsx:102]: Shall we wrap the returned element with brackets?

File: apps/admin-portal/src/components/identity-providers/wizard/steps/wizard-summary.tsx [hunk 7]
```diff
@@ -82,26 +82,28 @@ export const WizardSummary: FunctionComponent<WizardSummaryProps> = (
...
+    };
 
+    const getAuthenticatorComponent = () => {
+        return <>
             <Divider horizontal>Authenticator details</Divider>
 
             {authenticatorSummary && (
```
Comment (NipuniBhagya) [inline apps/admin-portal/src/components/identity-providers/wizard/steps/wizard-summary.tsx:109]: shall we get the circuit breaker to the next line?

File: apps/admin-portal/src/components/identity-providers/wizard/steps/wizard-summary.tsx [hunk 8]
```diff
@@ -117,6 +119,17 @@ export const WizardSummary: FunctionComponent<WizardSummaryProps> = (
...
+        </>;
+    };
+
+    const isAuthenticatorStepAvailable = () => {
+        return identityProvider?.federatedAuthenticators?.defaultAuthenticatorId;
+    };
+
+    return (
+        <Grid className="wizard-summary">
+            {getGeneralDetailsComponent()}
```
Comment (NipuniBhagya) [inline apps/admin-portal/src/components/identity-providers/wizard/steps/wizard-summary.tsx:131]: Let's keep a space before and after curly brackets. Please check other places as well.

File: apps/admin-portal/src/components/identity-providers/wizard/identity-provider-create-wizard.tsx (hunk 1 above)
Comment (tharindu-bandara) [reply apps/admin-portal/src/components/identity-providers/wizard/identity-provider-create-wizard.tsx]: Fixed in 6311ed7b87d8256e38389b5145f845d46e4c8564.

File: apps/admin-portal/src/components/identity-providers/wizard/steps/general-settings.tsx (hunk 2 above)
Comment (tharindu-bandara) [reply apps/admin-portal/src/components/identity-providers/wizard/steps/general-settings.tsx]: Fixed in 6311ed7b87d8256e38389b5145f845d46e4c8564.

File: apps/admin-portal/src/components/identity-providers/wizard/steps/wizard-summary.tsx (hunk 3 above)
Comment (tharindu-bandara) [reply apps/admin-portal/src/components/identity-providers/wizard/steps/wizard-summary.tsx]: Fixed in 6311ed7b87d8256e38389b5145f845d46e4c8564.

File: apps/admin-portal/src/components/identity-providers/wizard/steps/wizard-summary.tsx (hunk 4 above)
Comment (tharindu-bandara) [reply apps/admin-portal/src/components/identity-providers/wizard/steps/wizard-summary.tsx]: Fixed in 6311ed7b87d8256e38389b5145f845d46e4c8564.

File: apps/admin-portal/src/components/identity-providers/wizard/steps/wizard-summary.tsx (hunk 5 above)
Comment (tharindu-bandara) [reply apps/admin-portal/src/components/identity-providers/wizard/steps/wizard-summary.tsx]: Fixed in 6311ed7b87d8256e38389b5145f845d46e4c8564.

File: apps/admin-portal/src/components/identity-providers/wizard/steps/wizard-summary.tsx (hunk 6 above)
Comment (tharindu-bandara) [reply apps/admin-portal/src/components/identity-providers/wizard/steps/wizard-summary.tsx]: Fixed in 6311ed7b87d8256e38389b5145f845d46e4c8564.

File: apps/admin-portal/src/components/identity-providers/wizard/steps/wizard-summary.tsx (hunk 8 above)
Comment (tharindu-bandara) [reply apps/admin-portal/src/components/identity-providers/wizard/steps/wizard-summary.tsx]: Fixed in 6311ed7b87d8256e38389b5145f845d46e4c8564.

File: apps/admin-portal/src/components/identity-providers/wizard/steps/wizard-summary.tsx (hunk 7 above)
Comment (tharindu-bandara) [reply apps/admin-portal/src/components/identity-providers/wizard/steps/wizard-summary.tsx]: Fixed with aaccd78903a82df120e76ba561155546a19c6204.

**Pull request merged**